cpp/src/parquet/bloom_filter.cc

```diff
@@ -128,7 +190,7 @@ void BlockSplitBloomFilter::SetMask(uint32_t key, BlockMask& block_mask) const {
 
 bool BlockSplitBloomFilter::FindHash(uint64_t hash) const {
   const uint32_t bucket_index =
-      static_cast<uint32_t>((hash >> 32) & (num_bytes_ / kBytesPerFilterBlock - 1));
+      static_cast<uint32_t>(((hash >> 32) * (num_bytes_ / kBytesPerFilterBlock)) >> 32);
   uint32_t key = static_cast<uint32_t>(hash);
   uint32_t* bitset32 = reinterpret_cast<uint32_t*>(data_->mutable_data());
```

Comment:
This would be better as:
```suggestion
  const uint32_t* bitset32 = reinterpret_cast<const uint32_t*>(data_->data());
```